lib/Inspection/JsonPrintInspector.h

```diff
@@ -0,0 +1,263 @@
+    return {};
+  }
+
+  [[nodiscard]] Status::Success endField() { return {}; }
+
+  [[nodiscard]] Status::Success beginArray() {
+    _stream << '[' << _linebreak;
+    incrementIndentationLevel();
+    return {};
+  }
+
+  [[nodiscard]] Status::Success endArray() {
+    decrementIndentationLevel();
+    _stream << _linebreak << _indentation << ']';
+    return {};
+  }
+
+  template<class U>
+  struct FallbackContainer {
+    explicit FallbackContainer(U&&) {}
+  };
+  template<class Fn>
+  struct FallbackFactoryContainer {
+    explicit FallbackFactoryContainer(Fn&&) {}
+  };
+
+  template<class Invariant>
+  struct InvariantContainer {
+    explicit InvariantContainer(Invariant&&) {}
+  };
+
+ private:
+  JsonPrintInspector(std::ostream& stream, std::string indentation,
+                     Context const& context)
+      : Base(context), _stream(stream), _indentation(std::move(indentation)) {}
+
+  template<class>
+  friend struct detail::EmbeddedFields;
+  template<class, class...>
+  friend struct detail::EmbeddedFieldsImpl;
+  template<class, class, class>
+  friend struct detail::EmbeddedFieldsWithObjectInvariant;
+  template<class, class>
+  friend struct detail::EmbeddedFieldInspector;
+
+  template<class, class>
+  friend struct SaveInspectorBase;
+
+  using EmbeddedParam = std::monostate;
+
+  template<std::size_t Idx, std::size_t End, class T>
+  [[nodiscard]] auto processTuple(T const& data) {
+    if constexpr (Idx < End) {
+      _stream << _indentation;
+      return process(*this, std::get<Idx>(data))  //
+             | [&]() {
+                 if constexpr (Idx + 1 < End) {
+                   _stream << ',' << _linebreak;
+                 }
+                 return processTuple<Idx + 1, End>(data);
+               };
+    } else {
+      return Status::Success{};
+    }
+  }
+
+  template<class Iterator>
+  [[nodiscard]] auto processList(Iterator begin, Iterator end)
+      -> decltype(process(std::declval<JsonPrintInspector&>(), *begin)) {
+    for (auto it = begin; it != end;) {
+      _stream << _indentation;
+      if (auto res = process(*this, *it); !res.ok()) {
+        return res;
+      }
+      if (++it != end) {
+        _stream << ',' << _linebreak;
+      }
+    }
+    return {};
+  }
+
+  template<class T>
+  [[nodiscard]] auto processMap(T const& map)
+      -> decltype(process(std::declval<JsonPrintInspector&>(),
+                          map.begin()->second)) {
+    auto end = map.end();
+    _stream << _linebreak;
+    for (auto it = map.begin(); it != end;) {
+      auto&& [k, v] = *it;
+      _stream << _indentation << '"' << k << "\":" << _separator;
+      if (auto res = process(*this, v); !res.ok()) {
+        return res;
+      }
+      if (++it != end) {
+        _stream << ',' << _linebreak;
+      }
+    }
+    return {};
+  }
+
+  void incrementIndentationLevel() {
+    if (_format == JsonPrintFormat::kPretty) {
+      for (unsigned i = 0; i < IndentationPerLevel; ++i) {
```

Comment:
Please use append

Reply:
Good call, I did not know that there was an append overload for this.